### PR TITLE
Implement Visitor for SQL AST

### LINKED ISSUE #1511: Implement `Visitor` for SQL AST
Part of #1218

### PR DESCRIPTION
**Pull request by jackwener**
I hereby agree to the terms of the CLA available at: https://databend.rs/policies/cla/

## Summary

Implement Visitor for SQL AST.

Now, there is just `expression visitor`.

## Changelog

- New Feature

## Related Issues

Fixes #1511

Unit Tests

Stateless Tests

### CODE REVIEW

File: query/src/sql/parser/ast/expression_visitor.rs [hunk 1]
```diff
@@ -0,0 +1,107 @@
+use common_datavalues::DataType;
```
Comment (PsiACE) [inline query/src/sql/parser/ast/expression_visitor.rs:1]: pls add a license header

**Review by PsiACE — COMMENTED** [review]
also, run `make lint` locally

**Comment by jackwener** [comment]
OK

**Comment by BohuTANG** [comment]
Cool, it would be nice to add some tests :D

**Comment by leiysky** [comment]
@jackwener It's better to visit ast by reference but not moving, since a visitor shouldn't take ownership of ast.

File: query/src/sql/parser/ast/ast_visitor.rs [hunk 2]
```diff
@@ -0,0 +1,394 @@
...
+use crate::sql::parser::ast::SelectStmt;
+use crate::sql::parser::ast::SelectTarget;
+use crate::sql::parser::ast::SetExpr;
+use crate::sql::parser::ast::SetOperator;
+use crate::sql::parser::ast::Statement;
+use crate::sql::parser::ast::TableAlias;
+use crate::sql::parser::ast::TableReference;
+use crate::sql::parser::ast::TypeName;
+use crate::sql::parser::ast::UnaryOperator;
+
```
Comment (jackwener) [inline query/src/sql/parser/ast/ast_visitor.rs:37]: Maybe, there should be annotation here like `PlanVisitor`? @leiysky

**Comment by jackwener** [comment]
> Cool, it would be nice to add some tests :D

It is just default visitor, and don't have specific behavior. It may not need to test.

File: query/src/sql/parser/ast/ast_visitor.rs (hunk 2 above)
Comment (leiysky) [reply query/src/sql/parser/ast/ast_visitor.rs:37]: It's not required, you can suit yourself.

**Review by sundy-li — APPROVED** [review]
Let's merge it first.

**Comment by zhang2014** [comment]
@jackwener Thank you contribution.

**Pull request merged**